arangod/RocksDBEngine/RocksDBMethods.h

```diff
@@ -140,4 +142,30 @@ struct IndexingEnabler {
   RocksDBMethods* _meth;
 };
 
+class ConcurrencyControlToggler {
+ public:
+  ConcurrencyControlToggler(ConcurrencyControlToggler const&) = delete;
+  ConcurrencyControlToggler& operator=(ConcurrencyControlToggler const&) =
+      delete;
+
+  explicit ConcurrencyControlToggler(RocksDBMethods* meth, bool isExclusive)
+      : _meth(meth), _exclusive(isExclusive) {}
+
+  ~ConcurrencyControlToggler() {
+    if (!_exclusive) {
+      _meth->SetSkipConcurrencyControl(false);
+    }
+  }
+
+  void setConcurrencyControl(bool value) {
+    if (!_exclusive) {
+      _meth->SetSkipConcurrencyControl(!value);
+    }
+  }
+
+ private:
+  RocksDBMethods* _meth;
+  bool const _exclusive;
```

Comment:
We should either check that `_meth` is not null or use a reference instead. Also, can we call it `_methods` instead of `_meth? ;)

Reply:
I renamed the instance variable from `_meth` to `_methods` in 3 structs in that header.
I did not change its type from pointer to reference because in 2 of the 3 structs `_methods` can legitimately be a nullptr if the initial condition was false. Thus I added assertions to the constructor that the passed in constructor argument `methods` is always a non-nullptr.